torchvision/prototype/models/depth/stereo/crestereo.py

```diff
@@ -1055,11 +1057,145 @@ def _crestereo(
     return model
 
 
+_COMMON_META = {
+    "min_size": (384, 512),
+}
+
+
 class CREStereo_Base_Weights(WeightsEnum):
-    pass
+    """The metrics reported here are as follows.
+
+    ``mae`` is the "mean-average-error" and indicates how far (in pixels) the
+    predicted disparity is from its true value (equivalent to ``epe``). This is averaged over all pixels
+    of all images. ``1px``, ``3px``, ``5px`` and indicate the percentage of pixels that have a lower
+    error than that of the ground truth. ``relepe`` is the "relative-end-point-error" and is the
+    average ``epe`` divided by the average ground truth disparity. ``f1`` corresponds to the average of pixels whose epe
+    is either >3px, or whom's ``relepe`` is higher than 0.05 (therefore lower is better).
+
+    """
+
+    MEGVII = Weights(
+        # Weights ported from https://github.com/megvii-research/CREStereo
+        url="https://download.pytorch.org/models/crestereo_756c8b0f.pth",
+        transforms=StereoMatching,
+        meta={
+            **_COMMON_META,
+            "num_params": 5432948,
+            "recipe": "https://github.com/megvii-research/CREStereo",
+            "_metrics": {
+                "Middlebury2014-train": {
+                    # number of cascade stages
+                    1: {
+                        # number of refininement interations
+                        2: {
+                            "mae": 1.704,
+                            "rmse": 3.738,
+                            "1px": 0.738,
+                            "3px": 0.896,
+                            "5px": 0.933,
+                            "relepe": 0.157,
+                            "f1": 10.041,
+                        },
+                        5: {
+                            "mae": 0.956,
+                            "rmse": 2.963,
+                            "1px": 0.88,
+                            "3px": 0.948,
+                            "5px": 0.965,
+                            "relepe": 0.124,
+                            "f1": 5.056,
+                        },
+                        10: {
+                            "mae": 0.792,
+                            "rmse": 2.765,
+                            "1px": 0.905,
+                            "3px": 0.958,
+                            "5px": 0.97,
+                            "relepe": 0.114,
+                            "f1": 4.059,
+                        },
+                        20: {
+                            "mae": 0.749,
+                            "rmse": 2.706,
+                            "1px": 0.907,
+                            "3px": 0.961,
+                            "5px": 0.972,
+                            "relepe": 0.113,
+                            "f1": 3.782,
+                        },
+                    },
+                    2: {
```

Comment:
It seems to me the cascade=2 result is worse than cascade=1, in this case we may not need to put cascade=2 here? (just need to explain on the docs that this result is obtained with cascade=1)

Reply:
I have no clear answer for this solution, it boils down to how the users might be training their weights.

Reply:
Just curious, how should we train the model so that we can get better result in the cascaded inference?

Reply:
I am unaware if there is a recommended way of doing this given there's no hint of it neither in the paper nor the provided implementation.

My only remark so far is that cascaded inference can work better at very large resolution but that is not a given. Performance is tightly tied to the scene disparity distribution that is presented to the model throughout training.

Given that the model is not trained in cascaded manner, the attention operators at the lowest resolution become "overfitted" to the disparity distribution and aspect ratios encountered during training.

A `2X` change in resolution will most likely have a low impact on the evaluation given that for the base training setting resolution (`384x512`) the non-offseted correlation layer can account for over half of the entire image. (a pixel in the lowest feature map has a receptive `32x32`, and the default `1d search` is a window of `1x9`, therefore the horizontal search can account for `288` horizontal pixels in the original image).

If the offset values would push individual pixel by a value of `1` away from their previous centre (basically obtaining a dilation factor of `2`) we would end up with a correlation product that scanned almost the entire image horizontally. Hence, the importance of the training data distribution.

In this case, going up to a resolution of `768x1024` will not produce a significant change in the networks capabilities (as most of the training sets will not have pixels that are further away than a quarter of the entire image). As such, the cascaded inference can be seen as a way of bringing back the data in the original "search" distribution when faced with extremely large resolutions.

Reply:
To further exemplify the importance of training data one can improve the evaluation on `Middlebury2014` by almost `0.2 mae` with very minimal fine-tuning. Based on the datasets and scales which are fed into the model during fine-tuning, this improvement can appear after less than `10000` steps.